### PR TITLE
GH Actions: add linting for GH Action yaml files

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

There was not linter to protect from errors in GH Action yaml files

## How does this fix it?

Add `lint-gh-actions.yml` that runs linter to protect from errors in GH Action yaml files.
Also fix all the lint problems.

## How to test this PR
Ensure all GH Actions continue to work

### CODE REVIEW

File: .github/workflows/api-gateway-integration-test.yml [hunk 1]
```diff
@@ -113,7 +113,7 @@ jobs:
       - name: "API Gateway Rest Controller GET /hello - Validate Response"
         run: |
           # Validate Response:
-          if [ "${{ fromJson(steps.apiGatewayHello.outputs.response).data }}" != "Hi!"]; then
+          if [ "${{ fromJson(steps.apiGatewayHello.outputs.response).data }}" != "Hi!" ]; then
```
Comment (yoomlam) [inline .github/workflows/api-gateway-integration-test.yml:116]: fixed bug found by linter

File: .github/workflows/test-code.yml [hunk 2]
```diff
@@ -153,18 +153,18 @@ jobs:
...
 
-      - name: "Run contention classification tests"
+      - name: "Run Contention Classification tests"
         if: inputs.run_all_tests || steps.changed-files-specific.outputs.any_changed == 'true'
         run: |
           ./gradlew :domain-cc:test
 
       - name: "Run Employee Experience tests"
-        if: inputs.run_all_tests || steps.ee-changed-files-specific.any_changed  == 'true'
+        if: inputs.run_all_tests || steps.ee-changed-files-specific.outputs.any_changed  == 'true'
```
Comment (yoomlam) [inline .github/workflows/test-code.yml:168]: fixed bug found by linter

File: .github/workflows/lint-gh-actions.yml
Comment (yoomlam) [inline .github/workflows/lint-gh-actions.yml:1]: Crux of PR

File: .github/workflows/delete-old-releases.yml [hunk 3]
```diff
@@ -16,12 +16,12 @@ on:
       older_than:
         description: 'Delete pre-releases published older than this many days ago'
         required: true
-        type: number
+        type: string
```
Comment (yoomlam) [inline .github/workflows/delete-old-releases.yml:19]: There is no type `number`

**Review by tejans24 — APPROVED** [review]
LGTM

**Pull request merged**